R/mobr.R

```diff
@@ -320,7 +320,8 @@ rarefaction = function(x, method, effort=NULL, coords=NULL, latlong=NULL,
           coords = x_mob_in$spat
         }
     }
-    if (method == 'samp' | method == 'spat') {
+    if (method == 'spat') method = 'kNCN'
```

Comment:
Why would we only want 'kNCN' to be an option? Shouldn't we design this function so that various kinds of spatially constrained algos could be used? I could see two approaches to specfying this: 

1) change option `spat` to either `spat_kNCN` or `spat_euclid` 
2) keep `spat` as the value of `method` but add an argument for the kind of spatial algo. 

Thinking ahead I could see how the function rarefaction may evolve many different kinds of methods but the primary fundamental difference being if they are individual or sample-based. Then the second order difference is how those individuals or samples are accumulated (e.g., random, euclidean spatial, nearest neighbor, etc). This is fine but one immediate problem I see with my suggestion is that our "sample-based non-spatial rarefaction" is actually based off the individual based method (why does this have to be so confusing :P) So with all that in mind I currently favor option 1 (let me know what you think)

Reply:
Yes, I would keep both algorithms. I only substituted it here so it could be readily used with the rest of the package that uses `spat` as an argument (e.g. `get_delta_stats()`). 

I like the logic of option 2 that makes the distinction between our 3 basic curves. This would also mean that other functions such as `get_delta_stats()` can remain largely unchanged because they use `spat` as an input for `rarefaction()`. I would only add a `...` that passes on further arguments to `rarefaction()` which would need a `spat.algo` argument as you suggest. However, I would suggest we call the algorithms  `kNN` for k-nearest neighbor and `kNCN` for k-neirest centroid neighbor. This is the terminology used by Chiarucci et al. (2009). Eucleadian distances are used in both cases.